library/pk.c

```diff
@@ -1309,6 +1312,15 @@ int mbedtls_pk_sign_ext(mbedtls_pk_type_t pk_type,
         return MBEDTLS_ERR_PK_BAD_INPUT_DATA;
     }
 
+	if(pk_type == MBEDTLS_PK_RSASSA_PSS)
+	{
+		mbedtls_rsa_alt_context *rsa_pss = (mbedtls_rsa_alt_context *) ctx->pk_ctx;
+		int ret_pss = rsa_pss->sign_func(ctx, f_rng, p_rng, md_alg, hash_len, hash, sig);
+		if(ret_pss == 0) *sig_len = 384;
```

Comment:
This line is hardcoded which is not permitted. The information about DS key is stored in esp_ds_data_ctx_t where that struct is not visible here.